File: 89. Gray Code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        vector <int> res (1);
        for (int i = 0; i < n; i++)
        {
            for (int j = res.size()-1; j >= 0; j--)
                res.push_back(res[j] + pow(2, i));
        }
        return res;
    }
};

File: 89. Gray Code.py
class Solution(object):
    def grayCode(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        res = [0]
        for i in xrange(n):
            res += [j + pow(2,i) for j in res[::-1]]
        return res
